rocky-raft/src/rocky/raft/server/LeaderLogic.java

```diff
@@ -191,7 +234,7 @@ private void updateCommitIndex(ServerContext serverContext) throws IOException {
         for (int n = lastIndex; n > commitIndex; --n) {
             int count = 0;
             for (int i = 0; i < clusterSize; ++i) {
-                if (matchIndex[i] >= n) count++;
+                if (matchIndex.get(i) >= n) count++;
             }
 
             boolean majority = count > clusterSize / 2;
```

Comment:
We can't take a simple majority here because we need a majority from both the old and new configurations.